web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +193,456 @@ describe("<TaskHeader />", () => {
     });
   });
 
+  describe("tax registration modal", () => {
+    it("opens when tax task is marked complete", () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["START"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "START" })],
+        tasks: [taxTask],
+      });
```

Comment:
All of these new tests should be edited to remove all the unnecessary setup. Having extra setup that's not relevant to the outcome makes the tests hard to read and interpret for future users, and can be actively confusing because people will be trying to figure out the answers to those questions, like "wait why do we need a particular section here???" when the answer is, we don't.